client/rest-high-level/src/test/java/org/elasticsearch/client/MigrationIT.java

```diff
@@ -73,6 +77,16 @@ public void testUpgradeWithTaskApi() throws IOException, InterruptedException {
         awaitBusy(hasUpgradeCompleted);
     }
 
+    public void testGetDeprecationInfo() throws IOException {
+        createIndex("test", Settings.EMPTY);
+        DeprecationInfoRequest request = new DeprecationInfoRequest(Collections.singletonList("test"));
+        DeprecationInfoResponse response = highLevelClient().migration().getDeprecationInfo(request, RequestOptions.DEFAULT);
+        // a test like this cannot test actual deprecations
```

Comment:
This would be pretty easy to test in 6.x, but not so much in master. It might be worth adding a test case to 6.x, but I'm not sure.

Reply:
my fallback is that the response tests also test the response itself.

Reply:
I'll buy that argument.